src/pip/_internal/network/session.py

```diff
@@ -355,8 +355,9 @@ def __init__(
             # is typically considered a transient error so we'll go ahead and
             # retry it.
             # A 500 may indicate transient error in Amazon S3
+            # A 502 may be a transient error from a CDN
```

Comment:
This seems too vague. What CDN does this?

Reply:
Cloudflare and CloudFront both do this